### PR TITLE
verify that ArrayBuffer Content-Type is empty

### PR DESCRIPTION
**Pull request by igrigorik**
Discussion https://github.com/w3c/beacon/issues/40.

### CODE REVIEW

File: beacon/headers/header-content-type.html
Comment (cdumez) [inline beacon/headers/header-content-type.html]: As Anne commented on the bug:
""
Generally Content-Type's value is always a value that is completely known, other than the boundary parameter of multipart/form-data. It seems you could be much more strict overall here.
"""

Can we use assert_equals() in all cases and startsWith() only when the parameter is a FormData?

**Comment by igrigorik** [comment]
@cdumez yep, ptal. Confused myself earlier.. thanks for the nudge.

File: beacon/headers/header-content-type.html
Comment (cdumez) [inline beacon/headers/header-content-type.html]: Typo: Content-Type not referrer.

**Comment by cdumez** [comment]
LGTM otherwise.

**Comment by igrigorik** [comment]
Yikes, friday.. fixed.

**Comment by igrigorik** [comment]
@siusin can you please take a pass? Need your LGTM to land this.

**Comment by cdumez** [comment]
Oh, I thought I could approve :) I'll let @siusin approve then.

**Comment by igrigorik** [comment]
@cdumez if you have write access, you can.. use the GitHub review tool; I can't merge until I have the approve bit through GitHub :)

**Review by cdumez — APPROVED** [review]
LGTM

**Comment by igrigorik** [comment]
@cdumez awesome, thank you!

**Comment by cdumez** [comment]
Note that I am still not convinced the test is behaving as expected. The new version of the test still passes in WebKit even though I am pretty sure we are not sending an empty Content-Type for ArrayBuffer payloads.

I investigated a little in WebKit and I believe the issue is that the second test (the ArrayBuffer one) requires a CORS preflight. This is because the destination is cross-origin and the payload had a non-safelisted MIME type. I believe WebKit acts as per spec but please correct me if this is wrong.

Since the python script we're calling does not deal with OPTIONS requests, the preflight fails and we end up not sending the beacon. Apparently, when no beacon is received, the test considers the received Content-Type is the empty string so WebKit passes the check.

**Comment by cdumez** [comment]
An easy fix would be to use same origin in this test since we are mostly interested in Content-Type header, not CORS. We can test CORS separately (I actually have a couple of Beacon CORS tests downstream that I could contribute upstream).

I am happy to write a PR for switching the test to same origin if you agree.

**Comment by igrigorik** [comment]
@cdumez yep, that sounds reasonable, please do.. :)

**Comment by cdumez** [comment]
Ok, https://github.com/w3c/web-platform-tests/pull/6793

**Pull request merged**
